### PR TITLE
Uniform distribution of outcomes

### PR DESCRIPTION
**Pull request by hpmaxi**
Initial logic for #336

### CODE REVIEW

**Comment by fvictorio** [comment]
This will need to be a more complex than this.

First, this will be a checkbox that will disable the probability inputs. They will show `100/outcomes` as its probability, but actually their value won't matter.

When the initial funding is done, a `distributionHint` array is sent to the market. If the array is empty, then the probabilities are uniform. Plus, if the funding is done this way, the transaction costs less gas. So, even if the result is pretty similar to just sending `[100, 100, 100]` (or whatever), it's better to send `[]` for this gas reason.

**Comment by fvictorio** [comment]
Oh, another complexity is that if you disable the checkbox, then the resulting probabilities should be the same as the ones shown. Not a big deal, just something to keep in mind.

**Comment by hpmaxi** [comment]
I've repositioned the checkbox on another row to maintain the total probabilities position in the same column as the outcomes probabilities, but I didn't added colors neither a check/tick image.

File: app/src/components/common/outcomes/index.tsx [hunk 1]
```diff
@@ -173,9 +209,46 @@ const Outcomes = (props: Props) => {
     )
   }
 
+  const addNewOutcome = () => {
```
Comment (fvictorio) [inline app/src/components/common/outcomes/index.tsx]: There's a lot of duplication in these methods. A helper `uniform(outcomes)` should help.

File: app/src/components/common/outcomes/index.tsx [hunk 2]
```diff
@@ -209,45 +209,31 @@ const Outcomes = (props: Props) => {
     )
   }
 
+  const uniform = (outcomes: Outcome[]): Outcome[] => {
+    return outcomes.map(o => {
+      o.probability = 100 / outcomes.length
+      return o
```
Comment (fvictorio) [inline app/src/components/common/outcomes/index.tsx]: I'd rather avoid mutation here:

```
return outcomes.map(o => ({
  ...o,
  probability = 100 / outcomes.length
}))
```

(or something like that; didn't test it).

**Comment by gabitoesmiapodo** [comment]
Design is here: https://app.zenhub.com/workspaces/gnosis-conditional-exchange-5d71043b9b6cb500019ae2f7/issues/protofire/gnosis-conditional-exchange/336#issuecomment-575635633

**Comment by fvictorio** [comment]
I think we should just merge this and change the design in the next milestone. If having a raw HTML checkbox is too ugly, we can just hide it (or use a feature flag and hide it only in production).

**Comment by leolower** [comment]
@fvictorio I agree. The UI Design is going to change anyways so it is better to have this merged.

**Comment by fvictorio** [comment]
@mariano-aguero could you please review this? And if it's OK to you, then just merge it.

**Pull request merged**